lib/sender.js

```diff
@@ -256,7 +248,19 @@ function extractRecipient(recipient, callback) {
         return nextTick(callback, new Error("Recipient key '" + key + "' was provided as an incorrect type."));
     }
 
-    return nextTick(callback, null, value);
+    var param;
+
+    if (key === 'condition') {
+        param = 'condition';
+    }
+    else if (['registrationIds', 'registrationTokens'].indexOf(key) !== -1) {
+        param = 'registration_ids';
+    }
+    else {
+        param = 'to';
+    }
```

Comment:
I think I would extract this as a function:

```js
var param = getParamFromKey(key);

return nextTick(callback, null, value, param);
```

Reply:
@hypesystem Done, all tests pass.